lib/util/http/auth/jwt.go

```diff
@@ -0,0 +1,77 @@
+package auth
+
+import (
+	"crypto/rsa"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+//------------------------------------------------------------------------------
+
+// JWTConfig holds the configuration parameters for an JWT exchange.
+type JWTConfig struct {
+	Enabled       bool          `json:"enabled" yaml:"enabled"`
+	Claims        jwt.MapClaims `json:"claims" yaml:"claims"`
+	SigningMethod string        `json:"signing_method" yaml:"signing_method"`
+	PrivateKey    string        `json:"private_key" yaml:"private_key"`
+
+	// internal private fields
+	rsaKey *rsa.PrivateKey
+}
+
+// NewJWTConfig returns a new JWTConfig with default values.
+func NewJWTConfig() JWTConfig {
+	return JWTConfig{
+		Enabled:       false,
+		Claims:        nil,
+		SigningMethod: "",
+		PrivateKey:    "",
+	}
+}
+
+//------------------------------------------------------------------------------
+
+// Sign method to sign an HTTP request for an JWT exchange.
+func (j JWTConfig) Sign(req *http.Request) error {
+	if !j.Enabled {
+		return nil
+	}
+
+	// Must parse private key only once
+	if j.rsaKey == nil {
```

Comment:
This will need mutex locking as we can't guarantee `Sign` isn't called by parallel threads.